src/main/java/edu/rosehulman/quota/AddUserController.java

```diff
@@ -1,4 +1,32 @@
 package edu.rosehulman.quota;
 
-public class AddUserController {  
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+public class AddUserController implements Route {
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    String partnerId = request.params(":partnerId");
+    String productId = request.params(":productId");
+    String userId = request.params(":userId");
+
+    try {
+      // Add the user to our database
+      boolean added = Database.getInstance().addUser(partnerId, productId, userId);
+      if (!added) {
+        response.status(500);
+        return response;
+      }
+    } catch(Exception e) {
+      response.status(500);
+      return response;
+    }
```

Comment:
This entire try-catch can just be this:

```java
boolean added = Database.getInstance().addUser(partnerId, productId, userId);
if (!added) {
  throw new Exception();
}
```

When any exception is thrown, the server stops what it is doing and returns a pre-formatted internal server error to the client.